### PR TITLE
Add npm package machinery

### PR DESCRIPTION
**Pull request by imurchie**
First pass at making an npm package out of this. Once it is published it will take some work to get it wired in and finalized.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+const IOS = 'iOS';
+const TVOS = 'tvOS';
+
+const CARTHAGE_CMD = 'carthage';
+const CARTFILE = 'Cartfile.resolved';
+const CARTHAGE_ROOT = 'Carthage';
+
+async function hasTvOSSims () {
+  let devices = await getDevices();
+  devices = _.flatten(Object.values(devices)).filter((d) => d.isAvailable && d.platform === TVOS);
```
Comment (mykola-mokhnach) [inline index.js]: I assume `isAvailable` is only present since Xcode11

File: index.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  return !!devices.length;
```
Comment (mykola-mokhnach) [inline index.js]: _.isEmpty

File: gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  build: 'appium-webdriveragent',
+  projectRoot: __dirname,
+});
+
+
+gulp.task('clean:carthage', function cleanCarthage () {
+  return fs.rimraf('Carthage');
+});
+
+gulp.task('install:dependencies', gulp.series('transpile', 'clean:carthage', function installDependencies () {
```
Comment (mykola-mokhnach) [inline gulpfile.js:18]: we also usually execute xcodebuild with clean option to make sure the derived data caches are empty after WDA update

Comment (imurchie) [reply gulpfile.js:18]: I presume there is a bunch of stuff from `appium-xcuitest-driver` that ought to exist in this package. When wiring it in that will come out.

Comment (mykola-mokhnach) [reply gulpfile.js:18]: I don't have a full plan of what exactly should be moved out of xcuitest driver yet, but I believe we can discuss that later

Comment (imurchie) [reply gulpfile.js:18]: Me too.

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**